### PR TITLE
Add sample for data mapping

### PR DESCRIPTION
**Pull request by ShammiL**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: DataMapper/jsonResources/Customer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        "email": "johndoe@example.com",
+        "phone_number": "123-456-7890"
+    },
+    "address": {
+        "street": "123 Main St",
+        "city": "Anytown",
+        "state": "CA",
+        "postal_code": "12345",
+        "country": "United States"
+    }
```
Comment (SasinduDilshara) [inline DataMapper/jsonResources/Customer.json:15]:
```suggestion
    }
    
```

File: DataMapper/jsonResources/Customer.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
```
Comment (SasinduDilshara) [inline DataMapper/jsonResources/Customer.json]:
```suggestion
}

```

File: DataMapper/main.bal [hunk 3]
```diff
@@ -0,0 +1,75 @@
+type Contact record {
```
Comment (SasinduDilshara) [inline DataMapper/main.bal]: Shall we use closed records for this

File: DataMapper/main.bal [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    string email;
+    string phone_number;
+};
+
+type OrderData record {
+    string order_item_id;
```
Comment (SasinduDilshara) [inline DataMapper/main.bal]:
```suggestion
    string orderItemId;
```

File: DataMapper/main.bal [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    decimal price;
+    int stock_quantity;
+    string manufacturer;
+    string release_date;
+};
+
+type Customer record {
+    string name;
+    string email;
+    string phone_number;
```
Comment (SasinduDilshara) [inline DataMapper/main.bal]:
```suggestion
    string phoneNumber;
```

Comment (ShammiL) [reply DataMapper/main.bal]: This and above mentioned records are all generated.

File: DataMapper/main.bal [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+type Shipping_address record {
```
Comment (pcnfernando) [inline DataMapper/main.bal]:
```suggestion
type ShippingAddress record {
```

File: DataMapper/main.bal [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+type InventoryData record {
+    string product_id;
+    string name;
+    string description;
+    string category;
+    decimal price;
+    int stock_quantity;
+    string manufacturer;
+    string release_date;
```
Comment (pcnfernando) [inline DataMapper/main.bal]: let's use `releaseDate` format

File: DataMapper/main.bal (hunk 6 above)
Comment (ShammiL) [reply DataMapper/main.bal]: These records and their fields were generated by the Data mapper using this format.

File: DataMapper/main.bal (hunk 7 above)
Comment (ShammiL) [reply DataMapper/main.bal]: These records and their fields were generated by the Data mapper using this format.

Comment (LakshanWeerasinghe) [reply DataMapper/main.bal]: Changed the keys of the json files, and these got updated.

File: DataMapper/main.bal (hunk 6 above)
Comment (LakshanWeerasinghe) [reply DataMapper/main.bal]: Changed the keys of the json files, and these got updated.

File: DataMapper/jsonResources/Customer.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (pcnfernando) [inline DataMapper/jsonResources/Customer.json:1]: Refactor folder. name as resources/

File: DataMapper/Ballerina.toml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+[package]
+org = "etl"
+name = "DataMapper"
```
Comment (pcnfernando) [inline DataMapper/Ballerina.toml:3]: use the same convenstion for org and name and the sample folder name

**Review by pcnfernando — APPROVED** [review]
Suggestions will be handled separately. Hence approving

**Pull request merged**